### ISSUE TITLE
A non-interactive session should wait for a *remote* debugger when Wait-Debugger is called

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#5272]
Currently, when you invoke a script that contains a `Wait-Debugger` call (or for which breakpoints have been defined beforehand) via `pwsh -noninteractive`, the local debugger still kicks in _interactively_.

In a true non-interactive scenario - such as when `pwsh` is being called invisibly by another program such as Jenkins - the `Wait-Debugger` call is effectively ignored, preventing out-of-process debugging.

(Essentially, the in-process nested debug prompt tries to open and - due to not being able to receive interactive input - auto-closes, and script execution continues.)

The expected behavior is to block until an out-of-process debugger attaches, as is already the case when you invoke `Wait-Debugger` in the context of a _background job_.

Steps to reproduce
------------------

Assume the following script, `t.ps1`:

```powershell
Wait-Debugger
'after'
```

```powershell
pwsh -noninteractive -noprofile ./t.ps1
```

Expected behavior
-----------------

Script execution should block until a _remote_ (out-of-process) debugger attaches.

Actual behavior
---------------

An interactive local debug prompt is displayed, despite explicitly requesting non-interactivity:

```none
Entering debug mode. Use h or ? for help. 

At /Users/mklement/Desktop/pg/pg.ps1:20 char:1
+ 'after'
+ ~~~~~~~
[DBG]: PS /Users/mklement/Desktop/pg>>
```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-beta.9 on macOS 10.13
PowerShell Core v6.0.0-beta.9 on Ubuntu 16.04.3 LTS
PowerShell Core v6.0.0-beta.9 on Microsoft Windows 10 Pro (64-bit; v10.0.15063)
Windows PowerShell v5.1.15063.674 on Microsoft Windows 10 Pro (64-bit; v10.0.15063)
```

**Comment by fasteiner** [comment]
Unfourtunatley the same issue still exists in PowerShell 7.3.4 and Jenkins:

code:
```powershell
"Retrieving all expired AD user accounts that are enabled ..." | Add-LogToFile -filename $fileName -level INFO
$debug = [System.Convert]::ToBoolean($env:DEBUG)
if ($debug -eq $true) {
    Set-LogLevel -level DEBUG
    Wait-Debugger
}
$subjectprefix = (Get-4meObject -params $Params -resourceType "request_templates" -Id $env:TEMPLATE).subject
```

Log:
![image](https://github.com/PowerShell/PowerShell/assets/75947402/ebcac2c4-9ba3-4a9f-be54-4c0abc13cb04)

# Environment data
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

**Comment by JustinGrote** [comment]
This is still a valid issue. CC @StevenBucher98 for reopen.

**Comment by theJasonHelmick** [comment]
@mklement0  -- Thank you for this issue -- The Interactive WG is reviewing this proposal.  We agree that an option to connect to a non-interactive session using wait-debugger would be beneficial.  We are considering several options to address this and are spending more time discussing.

**Comment by SeeminglyScience** [comment]
The Interactive WG reviewed this issue. We decided we cannot change the default behavior as folks do currently utilize it do debug interactively (even if that seems contradictory) - even if we do agree that that would have been a better design.

In lieu of that, we recognize that there is not currently a way to work around this (without getting very hacky). I have opened #21392 which will at least provide a way to wait for attach.
